src/charm.py

```diff
@@ -294,6 +302,7 @@ def _fqdn(self) -> List[str]:
             f"{self.app.name}.{self._namespace}",
             f"{self.app.name}.{self._namespace}.svc",
             f"{self.app.name}.{self._namespace}.svc.cluster.local",
+            socket.getfqdn(),
```

Comment:
The Ingress library defaults to using the `fqdn`. Therefore, we should ensure it's added to the list of `fqdn`s.

Reply:
i'm not just asking this to be a jerk :D  -- i know this charm is running in a container, and likely this container doesn't have multiple nics so this isn't an issue.  Is there any RANDOM chance this charm would have multiple nics (say like using multus) and you'd pick the wrong fqdn.

I know juju has something called "spaces" which lets you choose which network a certain relation is connected to -- but i don't know how the ops framework deals with those spaces.

I say all that to say this -- this is probably gonna work right in 99.9% of the cases so nice job

Reply:
Yeah, I believe that particular use case is quite specific. Currently, I'm not overly concerned about it, but we should remain on the look out in case a customer does have that specific use case. Thank you for sharing your valuable insights @addyess!